src/sync/index.js

```diff
@@ -606,4 +613,74 @@ async function updateTopicBalance(topicAddress, db) {
   }
 }
 
-module.exports = startSync;
+async function listUnspentBalance() {
+  let result = [];
+  try {
+    result = await qclient.listUnspent();
+  } catch (err) {
+    logger.error(`ListUnspent: ${err.message}`);
+  }
+
+  const unspentAddressBalanceDict = {};
+  const unspentAddressArray = [];
+  _.forEach(result, (addressInfo) => {
+    const {
+      address,
+      amount,
+    } = addressInfo;
+    if (_.isEmpty(unspentAddressBalanceDict[address])) {
+      unspentAddressBalanceDict[address] = { qtum: amount };
+      unspentAddressArray.push(address);
+    } else {
+      unspentAddressBalanceDict[address].qtum += amount;
+    }
+  });
+
+  const addressBatches = _.chunk(unspentAddressArray, rpcBatchSize);
+  const unspentAddressBalanceArray = [];
+  const getBotBalancesPromise = new Promise(async (resolve) => {
+    sequentialLoop(addressBatches.length, async (loop) => {
+      const getBotBalancePromises = [];
+      _.map(addressBatches[loop.iteration()], async (address) => {
+        const getBotBalancePromise = new Promise(async (resolve) => {
+          let botBalance = null;
+          try {
+            const resp = await bodhiToken.balanceOf({
+              owner: address,
+              senderAddress: address,
+            });
+
+            botBalance = resp.balance.toString(10);
+          } catch (err) {
+            logger.error(`BalanceOf ${address}: ${err.message}`);
+            botBalance = '0';
+          }
+          unspentAddressBalanceDict[address].bot = botBalance;
+          resolve();
+        });
+
+        getBotBalancePromises.push(getBotBalancePromise);
+      });
+
+      await Promise.all(getBotBalancePromises);
+      loop.next();
+    }, () => {
+      _.forEach(unspentAddressBalanceDict, (value, key) => {
+        unspentAddressBalanceArray.push({
+          address: key,
+          qtum: value.qtum,
+          bot: value.bot,
+        });
+      });
+      resolve();
+    });
+  });
+  await getBotBalancesPromise;
+  return unspentAddressBalanceArray;
+}
+
+module.exports = {
+  startSync,
+  calculateSyncPercent,
+  listUnspentBalance,
```

Comment:
does this need to be exported? looks like its only being used in sync/index

Reply:
@dwalintukan nope, it is also used in `resolver.js`